### PR TITLE
implemented disallow-identifier-names

### PR DESCRIPTION
**Pull request by alawatthe**
Related to https://github.com/jscs-dev/node-jscs/pull/943. I screwed the rebase :weary: Sorry

### CODE REVIEW

File: lib/rules/disallow-identifier-names.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * var good = 1;
+ * object[fine] = 2;
+ * object.fine = 3;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var temp = 1;
+ * object[foo] = 2;
```
Comment (mrjoelkemp) [inline lib/rules/disallow-identifier-names.js]: `foo` should be quoted

File: lib/rules/disallow-identifier-names.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * ```js
+ * "disallowIdentifierNames": ['temp', 'foo']
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var good = 1;
+ * object[fine] = 2;
```
Comment (mrjoelkemp) [inline lib/rules/disallow-identifier-names.js]: `fine` should be quoted

**Comment by mrjoelkemp** [comment]
A test asserting case sensitivity would be great. In other words, there should be a test that shows that `FOO` does not result in an error.

File: lib/rules/disallow-identifier-names.js (hunk 1 above)
Comment (alawatthe) [inline lib/rules/disallow-identifier-names.js]: The rule currently does not prohibit ```object['foo']``` (therefore, the foo should not have quotes in the example).

That said, I agree that we want to prohibit ```object['foo']```, too. Let me implement it.

**Comment by mrjoelkemp** [comment]
@alawatthe Sorry this fell through the cracks. Great work here. Thanks for contributing. Landed!

I think this will be a popular rule.

**Pull request closed without merging**